opencog/guile/SchemeSmobDV.cc

```diff
@@ -0,0 +1,372 @@
+/*
+ * SchemeSmobDV.cc
+ *
```

Comment:
Although putting everything n the guile directory is in keeping with the historical style, its ... well, the goal was to not do this any more, and so `SchemePrimitive.h` and `SchemeModule.h` were invented to allow new guile modules to be easily created, and new API's to be defined. I would kind-of be a lot more comfortable if DV's were done that way, instead of adding them here.  (and maybe/probably TV and AV should be refactored to use `SchemePrimitive` etc. as well, but that's rather low priority).

See, for example: `persist/guile/PersistSCM.cc` which defines `fetch-atom` and etc. up at the top.  other, maybe more opaque examples:  `opencog/query/PatternSCM.cc` which defines `cog-bind-first-n` and `cog-satisfy` etc. also, `persist/gearman/DistSCM.cc`  In principle, files like `guile/RandGenSCM.cc` and `guile/LoggerSCM.cc` should live in thier own distinct directory, with their own CMakefile, but historical accident means they don't.

Reply:
I think having DV scheme bindings in it's own module makes sense. In fact, I think the whole code should be in its own library, especially since it's rather big and experimental at this point.

Could be under `opencog/distributional_value`, `opencog/atoms/distributional_value` or such, what you guys think?

Reply:
Yea I agree that moving it into its own directory makes sense. If we do choose `opencog/atoms/distributional_value` i think we should also move truth values to `opencog/atoms/truthvalue` which i think is the correct thing to do.